plugins/catalog-backend/src/database/types.ts

```diff
@@ -0,0 +1,53 @@
+/*
+ * Copyright 2020 Spotify AB
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import * as yup from 'yup';
+
+export type DatabaseComponent = {
+  id: string;
+  locationId?: string;
+  name: string;
+};
+
+export type AddDatabaseComponent = {
+  locationId?: string;
```

Comment:
I think we could require this for now right? Until we end up with other methods of ingesting components

Reply:
Left optional for now because you can delete locations, which reminds me, I didn't write the code to set this to null if that happens. Anyway, that actually tells me that what should be done (upcoming ticket!) is probably to never delete locations, since we want to track refresh events and errors etc. Would like to leave as is for now and address in that ticket